pkg/apiserver/controller/queue/queue.go

```diff
@@ -289,6 +272,32 @@ func CreateQueue(ctx *logger.RequestContext, request *CreateQueueRequest) (Creat
 		}
 	}
 
+	runtimeSvc, err := runtime.GetOrCreateRuntime(clusterInfo)
+	if err != nil {
+		ctx.Logging().Errorf("GlobalVCQueue create request failed. error:%s", err.Error())
+		ctx.ErrorCode = common.QueueResourceNotMatch
+		ctx.ErrorMessage = err.Error()
+		deleteErr := storage.Queue.DeleteQueue(request.Name)
+		if deleteErr != nil {
+			ctx.Logging().Errorf("delete request roll back db failed. error:%s", deleteErr.Error())
+		}
+		return CreateQueueResponse{}, err
+	}
+
+	// create namespace if not exist in cluster
+	k8sRuntime := runtimeSvc.(*runtime.KubeRuntime)
+	coreNs := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: request.Namespace,
+		},
+	}
+	if _, err = k8sRuntime.CreateNamespaceIfNotExist(coreNs, metav1.CreateOptions{}); err != nil {
```

Comment:
It might be better to change function name to `CreateNamespace`

Reply:
ok